web/modules/custom_weather_module/src/Plugin/Block/CustomWeatherBlock.php

```diff
@@ -37,30 +38,40 @@ public function build() {
       '#theme' => 'weather_theme',
       '#weather_header' => 'Weather now in ',
       '#user_location' => $this->userLocation(),
-      '#conditions' => $this->userWeather(),
-      '#icon' => 'https://cdn.weatherapi.com/weather/64x64/day/113.png',
+      '#conditions' => $this->userWeather()['conditions'],
+      '#icon' => $this->userWeather()['icon'],
     ];
   }
 
   /**
    * Private function for getting user IP.
    */
   private function userLocation() {
-    // $user_ip = \Drupal::request()->getClientIp();
     // @todo fix 2 problems here:
     // 1. rewrite $user_ip request with dependency injection
     // 2. make the function get proper user IP that could be geo located
     // random IP from the Netherlands is being used down here
-    $user_ip = '0.0.0.0';
+    // phpcs:ignore
+    $user_ip = \Drupal::request()->getClientIp();
     $client = new Client();
-    $request = $client->get('http://ip-api.com/json/' . $user_ip);
-    $response = json_decode($request->getBody(), TRUE);
-    if ($response['status'] == 'success') {
-      $location = $response['city'] . ", " . $response['country'];
+    try {
+      $request = $client->get('http://ip-api.com/json/' . $user_ip);
+      $response = json_decode($request->getBody(), TRUE);
+      if ($response['status'] == 'success') {
+        $location = $response['city'] . ", " . $response['country'];
+      }
+    }
+    catch (RequestException $e) {
+      // @todo use dependency injection
+      // phpcs:ignore
+      \Drupal::logger('custom_weather_module')->error('Message from @module: @message', [
+        '@module' => 'custom_weather_module',
+        '@message' => "GuzzleHttp\Exception\ConnectException: cURL error 6: Could not resolve host: ip-api.com",
+      ]);
```

Comment:
Do not forget that your method always should return the value to easy handling the result of work.
Hint: you can return NULL from the method and handle it with the '??' (Null Coalescing Operator)